File: 第一阶段/1.7-循环结构/Loops.cs
// C# 循环结构示例
// 对应文章：1.7 循环结构

using System;
using System.Linq;

// ==================== 传统 for 循环 ====================
Console.WriteLine("=== 传统 for 循环 ===");
for (int i = 0; i < 5; i++)
{
    Console.Write($"{i} ");  // 0 1 2 3 4
}
Console.WriteLine();

// ==================== foreach 循环 ====================
Console.WriteLine("\n=== foreach 循环 ===");
int[] numbers = { 1, 2, 3, 4, 5 };
foreach (var num in numbers)
{
    Console.Write($"{num} ");
}
Console.WriteLine();

// 倒序遍历
Console.Write("倒序: ");
for (int i = numbers.Length - 1; i >= 0; i--)
{
    Console.Write($"{numbers[i]} ");
}
Console.WriteLine();

// ==================== 带索引的遍历 ====================
Console.WriteLine("\n=== 带索引遍历 ===");
// C# 的 foreach 没有内置 index，需要手动维护或用 LINQ
for (int i = 0; i < numbers.Length; i++)
{
    Console.WriteLine($"  [{i}] = {numbers[i]}");
}
// LINQ 方式（C# 等价于 Python 的 enumerate）
foreach (var (num, i) in numbers.Select((n, i) => (n, i)))
{
    Console.WriteLine($"  第{i + 1}个: {num}");
}

// ==================== while 循环 ====================
Console.WriteLine("\n=== while 循环 ===");
int i2 = 0;
while (i2 < 5)
{
    Console.Write($"{i2} ");
    i2++;
}
Console.WriteLine();

// ==================== do-while 循环（Python 没有）====================
Console.WriteLine("\n=== do-while 循环 ===");
int j = 0;
do
{
    Console.Write($"{j} ");
    j++;
} while (j < 5);
Console.WriteLine();
// Python 模拟：while True + break

// ==================== break 和 continue ====================
Console.WriteLine("\n=== break 和 continue ===");
Console.Write("跳过 5，到 10 停止: ");
for (int k = 0; k < 20; k++)
{
    if (k == 5) continue;  // 跳过 5
    if (k == 10) break;    // 遇到 10 停止
    Console.Write($"{k} ");  // 0 1 2 3 4 6 7 8 9
}
Console.WriteLine();

// ==================== LINQ 查询（C# 独有的声明式循环） ====================
Console.WriteLine("\n=== LINQ 查询 ===");
var evenNumbers = numbers.Where(n => n % 2 == 0).ToList();
var doubled = numbers.Select(n => n * 2).ToList();
int sum = numbers.Sum();
Console.WriteLine($"偶数: [{string.Join(", ", evenNumbers)}]");
Console.WriteLine($"翻倍: [{string.Join(", ", doubled)}]");
Console.WriteLine($"求和: {sum}");
// Python 对应：列表推导式 [n for n in numbers if n % 2 == 0]

File: 第一阶段/1.7-循环结构/loops.py
# Python 循环结构示例
# 对应文章：1.7 循环结构

# ==================== for-in 循环 ====================
print("=== for-in 循环 ===")
# Python 只有一种 for 循环（for-in），不像 C# 有 for、foreach、do-while
for i in range(5):
    print(i, end=" ")  # 0 1 2 3 4
print()

# range() 的三种用法
print(f"range(5):       {list(range(5))}")        # [0, 1, 2, 3, 4]
print(f"range(2, 5):    {list(range(2, 5))}")     # [2, 3, 4]
print(f"range(0, 10, 2): {list(range(0, 10, 2))}")  # [0, 2, 4, 6, 8]

# ==================== 遍历列表 ====================
print("\n=== 遍历列表 ===")
numbers = [1, 2, 3, 4, 5]
for num in numbers:
    print(num, end=" ")
print()

# 倒序遍历
print("倒序:", end=" ")
for num in reversed(numbers):
    print(num, end=" ")  # 5 4 3 2 1
print()

# ==================== 带索引的遍历 ====================
print("\n=== 带索引遍历 ===")
# Python 用 enumerate()，C# 的 foreach 自带 index
for i, num in enumerate(numbers):
    print(f"  [{i}] = {num}")
# 也可以指定起始索引
for i, num in enumerate(numbers, start=1):
    print(f"  第{i}个: {num}")

# ==================== zip() 并行遍历 ====================
print("\n=== zip() 并行遍历 ===")
names = ["Alice", "Bob", "Charlie"]
scores = [95, 87, 92]
for name, score in zip(names, scores):
    print(f"  {name}: {score}")
# C# 对比：C# 的 Zip() 方法或 LINQ 的 Zip()

# ==================== while 循环 ====================
print("\n=== while 循环 ===")
i = 0
while i < 5:
    print(i, end=" ")
    i += 1
print()

# Python 没有 do-while，但可以模拟
print("模拟 do-while:", end=" ")
i = 0
while True:
    print(i, end=" ")
    i += 1
    if i >= 5:
        break  # 先执行一次，再判断

# ==================== break 和 continue ====================
print("\n\n=== break 和 continue ===")
print("跳过 5，到 10 停止:", end=" ")
for k in range(20):
    if k == 5: continue  # 跳过 5
    if k == 10: break    # 遇到 10 停止
    print(k, end=" ")    # 0 1 2 3 4 6 7 8 9
print()

# ==================== for-else（Python 独有！） ====================
print("\n=== for-else ===")
# 规则：循环正常结束（没有 break）才执行 else
# 适合搜索场景：找到就 break，没找到就执行 else

# 场景1：找到了，else 不执行
search_list = [1, 3, 5, 7, 9, 11]
target = 7
for item in search_list:
    if item == target:
        print(f"找到了 {target}")
        break
else:
    print(f"没找到 {target}")
# 输出：找到了 7

# 场景2：没找到，else 执行
target = 6
for item in search_list:
    if item == target:
        print(f"找到了 {target}")
        break
else:
    print(f"没找到 {target}")
# 输出：没找到 6

# ==================== 遍历字典 ====================
print("\n=== 遍历字典 ===")
person = {"name": "Alice", "age": 25, "city": "Beijing"}
for key, value in person.items():
    print(f"  {key}: {value}")
